pkg/deployer/deployer.go

```diff
@@ -35,31 +34,43 @@ func (d *Deployer) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine)
 
 // GetKubeConfig returns the kubeConfig after the bootstrap process is complete.
 func (d *Deployer) GetKubeConfig(cluster *clusterv1.Cluster, master *clusterv1.Machine) (string, error) {
-	ip, err := d.GetIP(cluster, master)
+
+	// Load provider config.
+	config, err := providerv1.ClusterSpecFromProviderSpec(cluster.Spec.ProviderSpec)
 	if err != nil {
-		return "", err
+		return "", errors.Errorf("failed to load cluster provider status: %v", err)
 	}
 
-	homeDir, ok := os.LookupEnv("HOME")
-	if !ok {
-		return "", fmt.Errorf("unable to use HOME environment variable to find SSH key: %v", err)
+	cert, err := certificates.DecodeCertPEM(config.CAKeyPair.Cert)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to decode CA Cert")
+	} else if cert == nil {
+		return "", errors.New("certificate not found in config")
+	}
+
+	key, err := certificates.DecodePrivateKeyPEM(config.CAKeyPair.Key)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to decode private key")
+	} else if key == nil {
+		return "", errors.New("key not found in status")
 	}
 
-	machineSpec, err := providerv1.MachineSpecFromProviderSpec(master.Spec.ProviderSpec)
+	ip, err := d.GetIP(cluster, master)
 	if err != nil {
 		return "", err
 	}
 
-	result := strings.TrimSpace(util.ExecCommand(
-		"ssh", "-i", homeDir+"/.ssh/openstack_tmp",
-		"-o", "StrictHostKeyChecking no",
-		"-o", "UserKnownHostsFile /dev/null",
-		"-o", "BatchMode=yes",
-		fmt.Sprintf("%s@%s", machineSpec.SshUserName, ip),
-		"echo STARTFILE; sudo cat /etc/kubernetes/admin.conf"))
-	parts := strings.Split(result, "STARTFILE")
-	if len(parts) != 2 {
-		return "", nil
+	server := fmt.Sprintf("https://%s:6443", ip)
```

Comment:
this might be the root cause for https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/399